### PR TITLE
[Android] Fix a few CloudRepository issues

### PR DESCRIPTION
**Pull request by darcywong00**
This PR fixes a few issues involving the Cloud Repository
1. Add some null checking to prevent crashes
2. This addresses an issue where a custom .kmp package is installed (which currently associates the first language listed in kmp.json). Trying to install additional languages listed in the kmp.json failed because the package ID info wasn't being propagated.

### User Testing
1. Install apk and launch Keyman
2. Install custom sil_tchad.kmp package (associates the first listed language Babalia Creole Arabic language)
3. Go to "Settings" --> Installed Languages, and cancel the keyboard catalog download
4. Hit the "plus" sign and select "Chadian Arabic" and install
5. Verify that language/keyboard pair is added

### CODE REVIEW

File: android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudRepository.java [hunk 1]
```diff
@@ -358,8 +358,9 @@ protected JSONObject wrapKmpKeyboardJSON(JSONArray languagesArray) {
 
   protected List<Keyboard> processKeyboardJSON(JSONObject query, boolean fromKMP) {
     List<Keyboard> keyboardsList = new ArrayList<>();
-    //keyboardModifiedDates = new HashMap<String, String>();
-
+    if (query.length() == 0) {
```
Comment (darcywong00) [inline android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudRepository.java]: This guards when query is `{}` and was causing `query.getJSONObject()` in line 368 to throw

**Comment by darcywong00** [comment]
@MakaraSok latest PR build available [here](https://build.palaso.org/repository/download/KeymanAndroid_TestPullRequests/170364:id/release/kMAPro-debug.apk)

**Comment by MakaraSok** [comment]
> ### User Testing
> 1. Install apk and launch Keyman
> 2. Install custom sil_tchad.kmp package (associates the first listed language Babalia Creole Arabic language)
> 3. Go to "Settings" --> Installed Languages, and cancel the keyboard catalog download
> 4. Hit the "plus" sign and select "Chadian Arabic" and install
> 5. Verify that language/keyboard pair is added

@darcywong00 The language pair got installed, but the keyboard does not seem to have the predictive text active right after the installation.

Tested on Android 6.0.1 using the artifact from this PR.

**Comment by MakaraSok** [comment]
This is working OK now that I installed both the language model and the custom keyboard provided. 

Passed! yay

**Pull request merged**